app/models/user.rb

```diff
@@ -73,4 +78,11 @@ def fetch_details(auth)
     self.remote_avatar_url = auth[:info][:image]
   end
 
+  def messages
+    Message.all_for_user(self).ordered
+  end
+
+  def companions(messages)
```

Comment:
this method should return users, not user_ids
and messages don't needed in params)